Lesson_3/Program.cs

```diff
@@ -0,0 +1,268 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Lesson_3
+{
+    class Program
+    {
+        private static double RiseToThePower_1(double num, int degree)
+        {
+            double result = 1;
+            for (int i = 1; i <= degree; i++)
+            {
+                result *= num;
+            }
+
+            return Math.Round(result, 2);
+        }
+
+        private static int[] RoundDigits_2(int devider)
+        {
+            int[] result = new int[1000];
+            int pointer = 0;
+            for (int i = 1; i < 1000; i++)
+            {
+                if (i % devider == 0)
+                {
+                    result[pointer] = i;
+                    ++pointer;
+                }
+            }
+
+            return result;
+        }
+
+        private static int TotalCountOfPositiveDigits_3(double maxNum)
+        {
+            int result = 0;
+            for (int i = 1; i <= (int)Math.Sqrt(maxNum); i++)
+            {
+                if (i * i < maxNum)
+                {
+                    ++result;
+                }
+            }
+
+            return result;
+        }
+
+        private static int Divider_4(int num)
+        {
+            int result = 0;
+            for (int i = 1; i <= num / 2; i++)
+            {
+                if (num % i == 0)
+                {
+                    result = i;
+                }
+            }
+
+            return result;
+        }
+
+        private static int SunOfDigitsDividedtoSeven_5(int start, int end)
+        {
+            int result = 0;
+
+            if (end > start)
+            {
+                for (int i = start; i < end; i++)
+                {
+                    if (i % 7 == 0)
+                    {
+                        result += i;
+                    }
+                }
+            }
+            else
+            {
+                for (int i = start; i > end; i--)
+                {
+                    if (i % 7 == 0)
+                    {
+                        result += i;
+                    }
+                }
+            }
+
+            return result;
+        }
+
+        private static int FibonacciLineToStep_6(int count)
```

Comment:
Wou wou, to complicated :)
Please google another solutions

Reply:
algorithm works in both ways, that's why it take more place. In case, if we will work only with positive numbers, it will be 2 times smaller.

Reply:
Only positive numbers. It's fibonacci task condition.